### DISCUSSION TITLE
"Visible Hair Accessory" component changes physbone properties/"Visible Hair Accessory"のComponentはphysboneのプロパティを変化します

### DISCUSSION THREAD

**Discussion by Freezorg** [bdunderscore/modular-avatar#626] [Q&A]
I have a relatively unmodified Wakahime Hitsuku avatar, and since her bangs don't really interfere with the field of vision, I wanted parts of her hair to be visible in first person. However, I found that when I applied MA Visible Hair Accessory to the hair root, the way the hair physbones moved changed drastically and became much more "floaty". To get around this, I applied the component to only the subparts of the hair I'd be able to see in first person anyway (on every subpart from Hair.Side.01.L to Hair.SideIn.01.R [here](https://i.imgur.com/oDzeLBv.png) ).

However, when I did this, all the parts with "back" in the name, the back of the hair, became immobile, as if the physbones have been disabled, yet still properly parented to and moving with the head.

When I made that second version I mentioned, I did take care to take the existing, no-anomalies avatar's project, copy it and make sure the ONLY changes were, in fact, the addition of the component to the subparts marked "side", followed by the build.

私が持ってる若姫ひつくのアバターは比較的未改造なのですが、前髪が視界の邪魔にならないのでファーストパーソンで髪の一部が見えるようにしたかったんです。しかし、HairRootにMA Visible Hair AccessoryのComponentを入れると、髪のフィズボーンの動きが大きく変わり、より「浮いた」感じになってしまうことがわかりました。これを回避するために、とにかくファースパーで見える髪の部分だけにコンポーネントを適用しました（[ここ](https://i.imgur.com/oDzeLBv.png) ではHair.Side.01.LからHair.SideIn.01.Rまでのすべての部分）。

しかし、そうすると、髪の後ろ側、名前に "back "がついているパーツはすべて、フィズボーンが無効化されたかのように動かなくなってしまいました。

2つ目のバージョンを作ったときは、既存の、異常のないアバターのプロジェクトをコピーし、唯一の変更が、「side」と書かれたサブパーツへのコンポーネントの追加と、それに続くビルドであることを確認しました。

**Comment by bdunderscore** [comment]
Unfortunately it's not possible to make just a portion of the bone chain visible - at least not without using _way too many_ constraints. I'm interested in the initial problem with the "floaty" hair though - can you show some screenshots of how the hair is configured and how the bones are laid out?

**Comment by Freezorg** [comment]
Sure!

Tree layout: [Unexpanded](https://i.imgur.com/oOk3hrp.png) [Expanded](https://i.imgur.com/V9nkwpx.png) (not full but... I think you get the picture)
General layout: [1](https://i.imgur.com/xBUjpgi.png) [2](https://i.imgur.com/BiWWJzm.png) [3](https://i.imgur.com/dNcgLhT.png) [4](https://i.imgur.com/mwz6JuY.png)

[This](https://files.catbox.moe/dhktji.mp4) video shows the exact parameters, before I add the MA Visible Hair Accessory component, of each group of bones (I scroll down to show which ones have the IsAnimated check, as well)

Here's how the bones behave with [no MA Visible Hair Component](https://files.catbox.moe/9wfe40.mp4) and [With MA Visible Hair Component on HairRoot](https://files.catbox.moe/mdzh6l.mp4) , the small strands at the top of the head are the most immediately obvious.

Probably unneeded but I also got a video of the [anomaly](https://files.catbox.moe/lemopn.mp4) when I set the component to only certain parts; it does work, but it seems to create a "duplicate" of the hair at the back that's static.

**Comment by bdunderscore** [comment]
Very strange. I wonder if there's some constraint evaluation order issues going on?

With the anomaly one, where precisely did you add the VHA component? It _should_ work if you put it at the same level as one of the VRCPhysBone components (but, then again, it should also work at the HairRoot level).

Also, is the base avatar on sale somewhere? It'd be easier to debug if I had a copy I could work with here.

That being said, I expect there might be major changes to the implementation of VHA once the announced [VRCHeadChop](https://ask.vrchat.com/t/developer-update-7-december-2023/21274#vrcheadchop-14) component is out, so I'll hold off on digging into this too much until that's out.

**Comment by Freezorg** [comment]
Like I said in the original post, I added the VHA component from Hair.Side.01.L to Hair.SideIn.01.R in the tree, including the ones between those.

The base avatar is sold on Booth, [Here](https://booth.pm/ja/items/4726958)
